fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+    .addUniqueIndex(IDX + "TENANT_KEY_SALT", TENANT_SALT)
+    .addUniqueIndex(IDX + "TENANT_KEY_TIDH", MT_ID, TENANT_HASH)
+    .addPrimaryKey("TENANT_KEY_PK", TENANT_KEY_ID)
+    .addForeignKeyConstraint(FK + TENANT_KEYS + "_TNID", adminSchemaName, TENANTS, MT_ID)
+    .setTablespace(fhirTablespace)
+    .build(model);
+```
+
+If this table is updated, the changes must be manually apply the migration steps to the `TENANT_KEYS` and the corresponding indices. 
+
+If the UniqueIndex or ForeignKeyConstraint is removed, the object must be removed. 
+
+## Managing PROCEDURE
+
+`FHIR_ADMIN.set_tenant` is created in the FHIR_ADMIN schema.  The stored procedure is a 
+
+If updates are made to the `set_tenant.sql`, the `FhirSchemaGenerator.buildAdminSchema`'s FhirSchemaConstants.INITIAL_VERSION must be updated to the next highest integer (in this case 2). 
+
+``` java
+// The set_tenant procedure can be created after all the admin tables are done
+ProcedureDef setTenant = model.addProcedure(this.adminSchemaName, 
+    SET_TENANT, 
+    FhirSchemaConstants.INITIAL_VERSION, 
+    () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, adminSchemaName, SET_TENANT.toLowerCase() + ".sql", null), 
+    Arrays.asList(allAdminTablesComplete), 
+    procedurePrivileges);
+        setTenant.addTag(SCHEMA_GROUP_TAG, ADMIN_GROUP);
+```
+
+When the `fhir-persistence-schema` updateProcedure is executed, the READ grant is applied again, and the procedure is updated. The Procedure privilege is reset upon re-executing the action. 
+
+If you change the stored procedure signature, you MUST drop the stored procedure, before applying the stored procedure to the database. 
+
+----------------------------------------------------------------
+
+# Schema: Tenant Data 
+
+The schema is based on the resources and types identified in the `ModelSupport.getResourceTypes()`.  Each resource has a set of supporting tables for FHIR Search Parameters. 
+
+> ![mt-table.png](mt-table.png)
+
+## Managing the TABLESPACE
+
+Each tenant receives a tenant specific tablespace. The tablespace is managed automatically with a default EXTENTSIZE of 128.  
+
+The schema is created in the multi-tenant schema `CreateSchemas` action, such as `FHIRDATA`.  When the schema is created a base schema is created with `UpdateSchema` action.  The `AllocateTenant` action provisions the schema in the tenant's table space for instance, `TS_TENANT1`.  
+
+Each tenant is allocated a partition based on the MT_ID and assigned to a tablespace. 
+
+The tenant tablespace is created one time and should not be changed, modified or migrated. 
+
+## Managing the SEQUENCES
+
+Each tenant has a two tenant specific sequences. These sequences are created a single time, and are NOT replaced or upgraded at any point in time. 
+
+| Sequence Name | Description |
+|----------|---------------------------------------------|
+| `fhir_sequence` | The logical id in the database, used for storing each FHIR Resource |
```

Comment:
Maybe a note to indicate that this is different from the resource logical ids?
Maybe something like this:
> Used to assign a database-internal id for each logical FHIR resource. This differs from the `Resource.id` associated with each resource in that its always database-assigned and unique across resource types.

Reply:
done